clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -133,6 +144,25 @@ public Struct toStruct() {
     }
 
     @Override
+    public String toString() {
```

Comment:
Thanks for the feedback @becketqin. Generating the string is not cheap and we don't need it unless debug logging is enabled. What do you think of the approach suggested here:

https://github.com/apache/kafka/pull/2689#discussion_r106285279